src/ompl/geometric/planners/quotientspace/algorithms/MultiQuotient.ipp

```diff
@@ -0,0 +1,322 @@
+    {
+        ompl::base::PlannerSolution psol(sol);
+        psol.setPlannerName(getName());
+        pdef_->addSolutionPath(psol);
+    }
+
+    return ompl::base::PlannerStatus::EXACT_SOLUTION;
+}
+
+template <class T>
+const ompl::base::ProblemDefinitionPtr &MultiQuotient<T>::getProblemDefinition(unsigned kQuotientSpace) const
+{
+    assert(kQuotientSpace >= 0);
+    assert(kQuotientSpace <= siVec_.size()-1);
+    return quotientSpaces_.at(kQuotientSpace)->getProblemDefinition();
+}
+
+template <class T>
+void MultiQuotient<T>::setProblemDefinition(const ompl::base::ProblemDefinitionPtr &pdef)
+{
+    this->Planner::setProblemDefinition(pdef);
+
+    //Compute projection of qInit and qGoal onto QuotientSpaces
+    ompl::base::Goal *goal = pdef_->getGoal().get();
+    ompl::base::GoalState *goalRegion = dynamic_cast<ompl::base::GoalState *>(goal);
+    double epsilon = goalRegion->getThreshold();
+    assert(quotientSpaces_.size() == siVec_.size());
+
+    ompl::base::State *sInit = pdef->getStartState(0);
+    ompl::base::State *sGoal = goalRegion->getState();
+
+    OMPL_DEVMSG1("Projecting start and goal onto QuotientSpaces.");
+
+    quotientSpaces_.back()->setProblemDefinition(pdef);
+
+    for (unsigned int k = siVec_.size()-1; k > 0 ; k--)
+    {
+        og::QuotientSpace *quotientParent = quotientSpaces_.at(k);
+        og::QuotientSpace *quotientChild = quotientSpaces_.at(k-1);
+        ompl::base::SpaceInformationPtr sik = quotientChild->getSpaceInformation();
+        ompl::base::ProblemDefinitionPtr pdefk = std::make_shared<base::ProblemDefinition>(sik);
+
+        ompl::base::State *sInitK = sik->allocState();
+        ompl::base::State *sGoalK = sik->allocState();
+
+        quotientParent->projectQ0Subspace(sInit, sInitK);
+        quotientParent->projectQ0Subspace(sGoal, sGoalK);
+
+        pdefk->setStartAndGoalStates(sInitK, sGoalK, epsilon);
+
+        quotientChild->setProblemDefinition(pdefk);
+
+        sInit = sInitK;
+        sGoal = sGoalK;
+    }
+}
+
+
+template <class T>
+void MultiQuotient<T>::getPlannerData(ompl::base::PlannerData &data) const
+{
+    unsigned int Nvertices = data.numVertices();
+    if (Nvertices > 0)
+    {
+        OMPL_ERROR("cannot get planner data if plannerdata is already populated");
+        OMPL_ERROR("PlannerData has %d vertices.", Nvertices);
+        exit(0);
+    }
+
+    unsigned int K = std::min(solutions_.size() + 1, quotientSpaces_.size());
+    K = std::min(K, stopAtLevel_);
+
+    for (unsigned int k = 0; k < K; k++)
+    {
+        og::QuotientSpace *Qk = quotientSpaces_.at(k);
+        Qk->getPlannerData(data);
+
+        // label all new vertices
+        unsigned int ctr = 0;
+        for (unsigned int vidx = Nvertices; vidx < data.numVertices(); vidx++)
+        {
+            ompl::base::PlannerDataVertexAnnotated &v = *static_cast<ompl::base::PlannerDataVertexAnnotated *>(&data.getVertex(vidx));
```

Comment:
Is it possible to do `static_cast<ompl::base::PlannerDataVertexAnnotated &>(data.getVertex(vidx))` instead?